HIP/hip-1056.md

```diff
@@ -1198,24 +1153,51 @@ message SingletonUpdateChange {
  * Values are generally full messages or byte arrays.
  */
 message MapUpdateChange {
+    /**
+      * A boolean flag indicating if the written value was identical to the original state value.
+      * <p>
+      * If this flag is set to true, it indicates that the value being
+      * written to the map is identical to the value that was already
+      * present in the map for the given key. This means that no actual
+      * change has occurred in the state of the map entry, and thus
+      * the state change can be considered a no-op for the purposes of
+      * state updates.
+      * <p>
+      * If this flag is set to false, it indicates that the value being
+      * written to the map is different from the value that was already
+      * present in the map for the given key. In this case, the state
+      * change represent in value represents an actual update to the map entry
+      * <p>
+      * This field is OPTIONAL. If not set, it is assumed to be false.
+      * <p>
+      * Note: This field is useful for optimization purposes, allowing
+      * clients to determine whether an update was necessary or if it
+      * was a no-op. It can help reduce unnecessary state updates and
+      * improve performance in scenarios where many updates are being
+      * made to the same map entry without any actual change in value.
+      * <p>
+      */
+    bool identical = 2;
```

Comment:
nit, since this is not final, and the fields are ordered alphabetically, we should also reassigned the filed number so this becomes 1, and `key` becomes 2.

*update*: the field numbers in the proto file are correct

Reply:
Fixed